graph-builder/src/graph.rs

```diff
@@ -19,11 +19,17 @@ use config;
 use failure::{Error, ResultExt};
 use registry;
 use serde_json;
-use std::collections::HashMap;
+use std::collections::{HashMap, HashSet};
 use std::sync::{Arc, RwLock};
 use std::thread;
 
 pub fn index(req: HttpRequest<State>) -> HttpResponse {
+    // Check for required client parameters.
+    let mandatory_params = &req.state().mandatory_params;
+    if commons::ensure_query_params(mandatory_params, req.query_string()).is_err() {
+        return HttpResponse::NotFound().finish();
```

Comment:
I think that "Bad Request" instead of "Not Found" would be more suitable

Reply:
404 is more specific but it doesn't fit this case. Unfortunately there's no more specific way than 400 to say "Bad Arguments", so I think "Bad Request" comes closest. On Linux I would compare this to EINVAL (not existent in HTTP protocol but 400 comes closest) vs ENOENT (compared to 404).